src/System.Device.Gpio/Interop/Unix/libgpiod/Interop.gpiod_chip_close.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+using System;
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    [DllImport("libgpiod", SetLastError = true)]
```

Comment:
>SetLastError = true [](start = 27, length = 19)

do we use last error when calling this method? if not, then it would be better to not use SetLastError

Reply:
this comment will apply to the rest of the interop methods.

---
In reply to: [244890760](https://github.com/dotnet/iot/pull/133#discussion_r244890760) [](ancestors = 244890760)

Reply:
I will remove from not used methods, and maybe remove all for better performance

Reply:
@tarekgh I actually don't know when it is needed and when it is not. Do you mind pointing out what setting that property will do?

---
In reply to: [244890852](https://github.com/dotnet/iot/pull/133#discussion_r244890852) [](ancestors = 244890852,244890760)

Reply:
This property is needed if the native call fails and want to get the native OS error. if the call of the pinvoke you can call Marshal.GetLastWin32Error to get such error. if you didn't add SetLastError = true, the error will be lost and you cannot get it. 

you need SetLastError = true only if you want to get the native error code. if you don't care about that, then you shouldn't add it.